graph/graph.go

```diff
@@ -213,11 +212,19 @@ func checkVisibility(target labels.Label, visibilities []string) bool {
 			return true
 		}
 
+		// We're cheating slightly by using buildtools' labels.Parse function to parse visibility
+		// identifiers, because "..." is permitted in a visibility identifier but not in build target
+		// names, but it'll work fine provided we handle the "..." case differently.
 		visibility := labels.Parse(v)
 
 		if filepath.Base(visibility.Package) == "..." {
 			pkg := filepath.Dir(visibility.Package)
-			if fs.IsSubdir(pkg, target.Package) {
```

Comment:
`fs.IsSubdir` is predominantly used for logic involving Go package names. Please package names don't work the same way (they can be empty, for instance), so I don't think it's a good idea to have the two sharing the same logic - hence `isSubpackage` below.